src/components/TimeInput.js

```diff
@@ -19,19 +19,24 @@ var TimeInput = React.createClass({
     }
   },
   propTypes: {
+    className: React.PropTypes.string,
     value: React.PropTypes.string,
     onChange: React.PropTypes.func,
     twelveHourTime: React.PropTypes.bool,
     defaultValue: React.PropTypes.string
   },
   render () {
+    let className = 'TimeInput'
+    if (this.props.className) {
+      className += this.props.className
+    }
     return (
-      <div className='TimeInput'>
+      <div className={className}>
         <input
           className='TimeInput-input'
           ref={(input) => { this.input = input }}
           type='text'
-          size={this.props.value.length - 1}
+          size={this.props.size}
```

Comment:
lets just delete this line, the component should not be concerned with controlling its own size